src/test/java/it/pagopa/transactions/commands/handlers/TransactionUserCancelHandlerTest.java

```diff
@@ -0,0 +1,166 @@
+import it.pagopa.ecommerce.commons.domain.v1.*;
+import it.pagopa.ecommerce.commons.v1.TransactionTestUtils;
+import it.pagopa.transactions.commands.TransactionUserCancelCommand;
+import it.pagopa.transactions.exceptions.AlreadyProcessedException;
+import it.pagopa.transactions.repositories.TransactionsEventStoreRepository;
+import it.pagopa.transactions.utils.TransactionsUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.*;
+
+@ExtendWith(MockitoExtension.class)
+public class TransactionUserCancelHandlerTest {
+
+    private TransactionUserCancelHandler transactionUserCancelHandler;
+    @Mock
+    private TransactionsEventStoreRepository<Void> transactionEventUserCancelStoreRepository;
+
+    @Mock
+    private QueueAsyncClient transactionUserCancelQueueClient;
+
+    private TransactionsEventStoreRepository<Object> eventStoreRepository = Mockito
+            .mock(TransactionsEventStoreRepository.class);
+    private final TransactionsUtils transactionsUtils = new TransactionsUtils(eventStoreRepository);
+
+    @BeforeEach
+    private void init() {
+        transactionUserCancelHandler = new TransactionUserCancelHandler(
+                transactionEventUserCancelStoreRepository,
+                transactionUserCancelQueueClient,
+                transactionsUtils
+        );
+    }
+
+    @Test
+    void shouldSaveAuthorizationEvent() {
+        String transactionId = TransactionTestUtils.TRANSACTION_ID;
+        TransactionUserCancelCommand transactionUserCancelCommand = new TransactionUserCancelCommand(
+                null,
+                new TransactionId(UUID.fromString(transactionId))
+        );
+
+        /* PRECONDITION */
+        Mockito.when(eventStoreRepository.findByTransactionId(transactionId))
+                .thenReturn((Flux) Flux.just(TransactionTestUtils.transactionActivateEvent()));
+
+        Mockito.when(transactionEventUserCancelStoreRepository.save(any()))
+                .thenAnswer(a -> Mono.just(a.getArgument(0)));
+
+        Mockito.when(transactionUserCancelQueueClient.sendMessageWithResponse(any(BinaryData.class), any(), any()))
+                .thenReturn(queueSuccessfulResponse());
+        /* EXECUTION TEST */
+        StepVerifier.create(transactionUserCancelHandler.handle(transactionUserCancelCommand))
+                .consumeNextWith(
+                        next -> {
+                            assertEquals(TransactionEventCode.TRANSACTION_USER_CANCELED_EVENT, next.getEventCode());
+                        }
+                )
+                .verifyComplete();
+    }
```

Comment:
Add some assertion here such as verify that eventStore, queue client etc are called once